csrc/ir/builder.cpp

```diff
@@ -21,6 +21,9 @@ Val* IrBuilder::newScalar(DataType dtype) {
   if (isPointerType(dtype)) {
     return IrBuilder::create<Int>(dtype);
   }
+  if (!std::holds_alternative<PrimDataType>(dtype.type)) {
```

Comment:
What case is this supposed to handle? I guess we are creating a scalar of a non-primitive data type?

Reply:
Yes, we are creating scalar with data type `ArrayOf`